homeassistant/components/sensor/sense.py

```diff
@@ -150,3 +198,59 @@ def update(self):
             state = self._data.get_trend(self._sensor_type,
                                          self._is_production)
             self._state = round(state, 1)
+
+
+class SenseDevice(Entity):
+    """Implementation of a Sense detected device."""
+
+    def __init__(self, realtime_data_wrapper, name, location, icon):
+        """Initialize the sensor."""
+        self._realtime_data_wrapper = realtime_data_wrapper
+        self._device_name = name
+        if location != '':
+            self._device_name = '{} ({})'.format(name, location)
+        self._name = '{} Energy Usage'.format(self._device_name)
+        self._icon = icon
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def force_update(self):
+        """Return if an same sensor value should still count as an update."""
+        return True
```

Comment:
Please remove `force_update` as this is not the case. You are actually throttling the updates and so it should not write it every time to the state machine.

Reply:
If you want force_update to be set to True, you need to define a SCAN_INTERVAL, that way Home Assistant will wait SCAN_INTERVAL time between calling update, which means there is new data.

Reply:
Thanks for the information. I'm still a little confused trying to understand the right thing to do, though. The reason I added `force_update` was to fix an issue where consecutive `0` state values were not seen as updates to state. This resulted in incorrect graphs, like this:
<img width="356" alt="screen shot 2018-02-19 at 7 20 16 pm" src="https://user-images.githubusercontent.com/382859/38469236-f3e485a2-3b16-11e8-966e-2fbb42548316.png">

With `force_update`, I get a more accurate graph, like this:
<img width="352" alt="screen shot 2018-04-08 at 10 24 22 am" src="https://user-images.githubusercontent.com/382859/38469247-11a0d51e-3b17-11e8-8945-26e168cc4aa6.png">

So if I include a `SCAN_INTERVAL`, does that mean I also should get rid of the existing throttling?

Thanks for your help.

Reply:
@balloob -- I know you are busy, so I'm sorry to ping you. Just still stumped about how to fix the issue above without using force_update. Thanks again for your help.

Reply:
You should get rid of throttling if you want to use force_update=True, yes. 

Because what happens with throttling is that you won't fetch new data, but by using force-update you are telling Home Assistant that there is new data, it's just the same.